test/unit/organisation_test.rb

```diff
@@ -14,14 +14,47 @@ def test_build_and_parse_xml
 
     # Build a new account from the XML
     result_org = XeroGateway::Organisation.from_xml(org_element)
-    
+
     # Check the account details
     assert_equal org, result_org
   end
-  
-  
+
+  def test_organisation_with_addresses
+    org = create_test_organisation
+    org.add_address(
+      :address_type => 'POBOX',
+      :line_1 => 'NEW LINE 1',
+      :line_2 => 'NEW LINE 2',
+      :line_3 => 'NEW LINE 3',
+      :line_4 => 'NEW LINE 4',
+      :city => 'NEW CITY',
+      :region => 'NEW REGION',
+      :post_code => '5555',
+      :country => 'Australia'
+    )
+
+    org_as_xml = org.to_xml
+
+    assert org_as_xml.include?("<Addresses><Address>")
+    assert org_as_xml.include?("NEW REGION")
+
+    org_element = REXML::XPath.first(REXML::Document.new(org_as_xml), "/Organisation")
+    result_org = XeroGateway::Organisation.from_xml(org_element)
+
+    assert_equal org, result_org
+  end
+
+  test "should raise if you add an unsupported type as an address" do
+    org = create_test_organisation
+    org.addresses = [ { :a => 123 }]
+
+    assert_raises "UnsupportedAttributeType" do
+      org.to_xml
+    end
+  end
+
```

Comment:
Have added a test, not sure it does but maybe I'm missing the scenario you're thinking of?